sysbuild/CMakeLists.txt

```diff
@@ -504,21 +504,8 @@ function(${SYSBUILD_CURRENT_MODULE_NAME}_pre_cmake)
     set_config_bool(mcuboot CONFIG_PCD_APP n)
   endif()
 
-  if(SB_CONFIG_BT_FAST_PAIR)
-    set_config_bool(${DEFAULT_IMAGE} CONFIG_BT_FAST_PAIR y)
-
-    if(DEFINED FP_MODEL_ID AND DEFINED FP_ANTI_SPOOFING_KEY)
-      if(SB_CONFIG_PARTITION_MANAGER)
-        include(${ZEPHYR_NRF_MODULE_DIR}/cmake/sysbuild/fast_pair_hex.cmake)
-      endif()
-
-      set(FP_DATA_PRESENT "y" CACHE INTERNAL "Fast Pair provisioning data provided" FORCE)
-    else()
-      message(WARNING "Fast Pair support is enabled but `FP_MODEL_ID` or `FP_ANTI_SPOOFING_KEY` were not provided, this is likely to cause a build error")
-      set(FP_DATA_PRESENT "n" CACHE INTERNAL "Fast Pair provisioning data provided" FORCE)
-    endif()
-  else()
-    set_config_bool(${DEFAULT_IMAGE} CONFIG_BT_FAST_PAIR n)
+  if(SB_CONFIG_BT_FAST_PAIR_PROV_DATA AND SB_CONFIG_PARTITION_MANAGER)
```

Comment:
Consider commenting in both places why do we introduce Fast Pair provisioning data generation in 2 various places (to simplify reading this CMakeLists.txt)